cocos/2d/CCNode.cpp

```diff
@@ -950,6 +958,8 @@ void Node::addChildHelper(Node* child, int localZOrder, int tag, const std::stri
     child->setParent(this);
     child->setCameraMask(getCameraMask());
 
+    child->_setOrderOfArrival(s_globalOrderOfArrival++);
```

Comment:
Well, it's ok, and Node::_setOrderOfArrival() may be needs change to Node::_increaseOrderOfArrival() or Node::_updateOrderOfArrival